bson/bson_binary_vector_spec_test.go

```diff
@@ -82,21 +81,18 @@ func TestBsonBinaryVectorSpec(t *testing.T) {
 	})
 }
 
-func convertSlice[T int8 | float32 | byte](s []interface{}) []T {
+func convertSlice[T int8 | float32 | byte](t *testing.T, data []byte) []T {
+	if len(data) == 0 {
+		return nil
+	}
+	var s []float64
+	err := UnmarshalExtJSON(data, true, &s)
+	if err != nil {
+		t.Fatalf("got %q while handling %s", err, string(data))
+	}
```

Comment:
Suggest using the `require` package 
```suggestion
	require.NoError(t, err)
```